### DISCUSSION TITLE
Permissions on methods of contracts

### DISCUSSION THREAD

**Discussion by izyak** [icon-project/IBC-Integration#305] [Ideas]
We have methods on contract like `createClient`, which can be initialized with wrong data by some external party. It makes sense to impose some restriction on this method.

Similarly, on the ICON side, since the handshaking for channel and connection can emit BTP Blocks w/o much cost, we need to think of a way to prevent that. For this, we can either have permissioned methods, or we need to charge a certain amount of fee from the caller, so unnecessary BTP Blocks would not be generated.

**Comment by ibrizsabin** [comment]
I think in case of iarchway host  we can make create client  and upgrade client to be permissioned and other methods can be left open ao that any relayer can connect and send messages.
